language_service/src/completion.rs

```diff
@@ -597,22 +589,48 @@ impl CompletionListBuilder {
             ItemKind::Namespace(namespace, _)
                 if !namespace.starts_with_sequence(&["Microsoft", "Quantum", "Unstable"]) =>
             {
-                let label = format!(
-                    "{}{}",
-                    if let Some(ref alias) = package_alias {
-                        format!("{alias}.")
-                    } else {
-                        String::new()
-                    },
-                    namespace.name()
-                );
+                let qualification = namespace
+                    .str_iter()
+                    .into_iter()
+                    .map(Rc::from)
+                    .collect::<Vec<_>>();
+                let label = format_external_name(&package_alias, &qualification[..], None);
                 Some(CompletionItem::new(label, CompletionItemKind::Module))
             }
             _ => None,
         })
     }
 }
 
+/// Format an external fully qualified name
+/// This will prepend the package alias and remove `Main` if it is the first namespace
+fn format_external_name(
+    package_alias_from_manifest: &Option<Arc<str>>,
+    qualification: &[Rc<str>],
+    name: Option<&str>,
+) -> String {
+    let mut fully_qualified_name: Vec<Rc<str>> = if let Some(alias) = package_alias_from_manifest {
+        vec![Rc::from(&*alias.clone())]
+    } else {
+        vec![]
+    };
+
+    if package_alias_from_manifest.is_some()
+        && qualification.len() == 1
+        && qualification.first() == Some(&"Main".into())
+    {
+        fully_qualified_name.append(&mut qualification[1..].to_vec());
```

Comment:
Since this is already predicated on qualification.len() == 1, this line will do nothing. I think you can just append the full qualification slice when the above condition is false.